codegen/README.md

```diff
@@ -0,0 +1,21 @@
+# Codegen
+
+## Generate
+
+### Generate service config files
+
+Run the following command to generate service config files:
+
+```bash
+pdm genconfig
```

Comment:
I'm opposed to installing tools like `pdm`. In fact, I want to avoid adding any new dependencies. We don't need to package/release/dist it. I expect this code generation can be executed with a simple `python3 xxx.py`, just like all other scripts. Thus I prefer to keep it inside `scripts`.

Or we can make `scripts` as a whole new package that managed as a real python project.

Reply:
> I want to avoid adding any new dependencies.

This is strange. It means we regenerate many utilities by hand.

Code generator isn't such a trivial thing.

Reply:
> Or we can make scripts as a whole new package that managed as a real python project.

This is possible. Or call it `tools`/`pytools`. I suppose we'd have some handy scripts may be not in Python (or Python is not a good fit).

Firstly I'd make a consensus that write a code generator without any 3rdparty will end up with write every utility by hand and it's even .. stupid.

Then we back to the naming issue about how we call the current "codegen" folder.

Reply:
> Firstly I'd make a consensus that write a code generator without any 3rdparty will end up with write every utility by hand and it's even .. stupid.

Agrees.

> Then we back to the naming issue about how we call the current "codegen" folder.

We can merge current `scripts` and `codegen` into one `tools` or `odev`.

And maybe we can use the same entry point: `odev`?

Reply:
> And maybe we can use the same entry point: `odev`?

This is much furthur on current codebase. We can start with `tools` as a python project first.